src/main/java/com/google/api/codegen/util/NameFormatterMixin.java

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+/**
+ * NameFormatterMixin is an abstract class that implements NameFormatter and
+ * simply forwards calls to another NameFormatter. This allows a child class
+ * to provide the interface of NameFormatter along with additional functionality.
+ *
+ * Note to future maintainers: This class should only contain methods which
+ * forward on to NameFormatter and nothing else; otherwise, it is no longer
+ * functioning in spirit as a mix-in.
+ */
+public abstract class NameFormatterMixin implements NameFormatter {
```

Comment:
This might more accurately be described as a [Proxy](https://en.wikipedia.org/wiki/Proxy_pattern). Mixin conveys the wrong thing to me, because I can't mix anything else into a `NameFormatterMixin`, due to Java's ban on multiple inheritance (at least until we move to Java 8).